witness/src/Dscp/Witness/Retranslate/Transaction.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE OverloadedLists #-}
+
+module Dscp.Witness.Retranslate.Transaction
+    ( TxRetranslator (..)
+    , makeRetranslator
+    ) where
+
+import qualified Control.Concurrent.STM as STM
+
+import qualified Data.HashMap.Strict as HashMap
+
+import Loot.Base.HasLens (HasLens (..))
+import Loot.Log (logError)
+
+import Dscp.Crypto (hash)
+import Dscp.Network.Wrapped (Listener (..), Worker (..), cliRecvUpdate, msgType, servPub, subType)
+import Dscp.Witness.Launcher.Mode (WitnessWorkMode)
+import Dscp.Witness.Mempool (MempoolVar, addTxToMempool, isInMempool)
+import Dscp.Witness.Messages (PubTx (..), SendTx (..))
+
+-- | Holds input queue for 'SendTx's and all moving parts of the retranslator.
+--
+--   'Worker's/'Listener's are there because we might at some point want to stop
+--   this mechanism gracefully.
+--
+--   They could be replaced by some `trStop :: m ()` method.
+--
+data TxRetranslator m
```

Comment:
How do you intend to launch/use it? I recommend to get rid of this module and add listeners/workers to `Listeners` and `Workers`. Queue should be wrapped in newtype and put into the witness context.